### PR TITLE
added multiple sql queries to speed up the report

### PR DESCRIPTION
**Pull request by IshikaM19**
- added multiple SQL queries instead of one large one 
- Speeds up the making of the report by a little bit

### CODE REVIEW

File: report/report.py [hunk 1]
```diff
@@ -11,6 +11,9 @@
 import boto3
 from botocore.exceptions import ClientError
 from xhtml2pdf import pisa
+from time import perf_counter
+
+# pylint: disable=E1136
```
Comment (caithomas6789) [inline report/report.py:17]: Maybe try to fix the pylint problem so you don't have to disable pylint.

File: report/report.py [hunk 2]
```diff
@@ -107,17 +80,45 @@ def create_table_three_columns(column_1: str, column_2: str, column_3: str, data
...
+                    SELECT sale_id, amount, item_id, country_id
+                    FROM sale_event
+                    WHERE DATE(sale_time) = CURRENT_DATE - INTERVAL '1 day';""")
+        tuples = curr.fetchall()
+        column_names = ['sale_id', 'amount', 'item_id',
+                        'country_id']
+
+        df = pd.DataFrame(tuples, columns=column_names)
+
+        return df
```
Comment (caithomas6789) [inline report/report.py:110]: Nice small SQL query which speeds up the script.

File: report/report.py [hunk 3]
```diff
@@ -107,17 +80,45 @@ def create_table_three_columns(column_1: str, column_2: str, column_3: str, data
...
     """
 
+    sale_event_data = load_sale_event_data(db_connection)
     # amount number of sales
-    amount_sales = data['sale_id'].nunique()
+    amount_sales = ('{:,}'.format(sale_event_data['sale_id'].nunique()))
 
     # amount income
-    amount_income = (data['amount'].sum())/100
+    amount_income = ('{:,}'.format((sale_event_data['amount'].sum())/100))
```
Comment (caithomas6789) [inline report/report.py:124]: Formatting will help make the report more readable 👍

File: report/report.py [hunk 4]
```diff
@@ -133,230 +134,468 @@ def get_key_analytics(data: pd.DataFrame) -> str:
...
+                    GROUP BY artist.artist_name
+                    ORDER BY amount DESC
+                    LIMIT 5;""")
+        tuples = curr.fetchall()
+        column_names = ['amount',
+                        'artist']
+
+        df = pd.DataFrame(tuples, columns=column_names)
+
+        return df
```
Comment (caithomas6789) [inline report/report.py:191]: These queries look good for finding the top 5 most popular artists.

File: report/report.py [hunk 5]
```diff
@@ -133,230 +134,468 @@ def get_key_analytics(data: pd.DataFrame) -> str:
...
 
-    unique_sales = data.drop_duplicates(subset='sale_id', keep='first')
-    popular_artists = unique_sales['artist'].value_counts().sort_values(ascending=False).head(
-        5).reset_index()
+    artist_data = load_top_5_popular_artist_data(db_connection)
 
-    artists = popular_artists.to_dict('records')
+    artists = artist_data.to_dict('records')
+
+    format_all_numbers(artists, 'count')
```
Comment (caithomas6789) [inline report/report.py:204]: Good formatting.

File: report/report.py [hunk 6]
```diff
@@ -133,230 +134,468 @@ def get_key_analytics(data: pd.DataFrame) -> str:
...
-        artist_sales/100).sort_values(ascending=False).head(5).reset_index()
+    artist_data = load_top_5_grossing_artist_data(db_connection)
+
+    artist_data['amount'] = (
+        artist_data['amount']/100)
 
-    artists = artist_sales.to_dict('records')
+    artists = artist_data.to_dict('records')
+
+    format_all_numbers(artists, 'amount')
```
Comment (caithomas6789) [inline report/report.py:225]: More good formatting.

File: report/report.py [hunk 7]
```diff
@@ -133,230 +134,468 @@ def get_key_analytics(data: pd.DataFrame) -> str:
...
 
-    unique_sales = data.drop_duplicates(subset='sale_id', keep='first')
-    album_sales = unique_sales.drop(
-        unique_sales[unique_sales['item_type'] == 'track'].index)
-    popular_albums = album_sales['item_name'].value_counts().sort_values(ascending=False).head(
-        5).reset_index()
+    albums = album_data.to_dict('records')
 
-    albums = popular_albums.to_dict('records')
+    format_all_numbers(albums, 'count')
```
Comment (caithomas6789) [inline report/report.py:296]: Even more good formatting.

File: report/report.py [hunk 8]
```diff
@@ -133,230 +134,468 @@ def get_key_analytics(data: pd.DataFrame) -> str:
...
+                    AND item_type.item_type_id = 2
+                    GROUP BY item.item_name
+                    ORDER BY amount DESC
+                    LIMIT 5;""")
+        tuples = curr.fetchall()
+        column_names = ['amount', 'item_name']
+
+        df = pd.DataFrame(tuples, columns=column_names)
+
+        return df
```
Comment (caithomas6789) [inline report/report.py:372]: These look like good queries to retrieve the data

File: report/report.py [hunk 9]
```diff
@@ -133,230 +134,468 @@ def get_key_analytics(data: pd.DataFrame) -> str:
...
-    track_sales = unique_sales.drop(
-        unique_sales[unique_sales['item_type'] == 'album'].index)
-    track_sales = track_sales.groupby(
-        'item_name')['amount'].sum()
-    track_sales = (
-        track_sales/100).sort_values(ascending=False).head(5).reset_index()
+    tracks = track_data.to_dict('records')
 
-    tracks = track_sales.to_dict('records')
+    format_all_numbers(tracks, 'amount')
```
Comment (caithomas6789) [inline report/report.py:407]: Good formatting.

File: report/report.py [hunk 10]
```diff
@@ -417,19 +657,19 @@ def generate_html_string(data: pd.DataFrame) -> str:
...
             </div>
             <h3 class="subtitle"> Tracks </h3>
             <div class="row">
                 <div class="column">
-                    {get_top_5_sold_tracks(data)}
+                    {get_top_5_sold_tracks(db_connection)}
                 </div>
                 <div class="column">
-                    {get_top_5_grossing_tracks(data)}
+                    {get_top_5_grossing_tracks(db_connection)}
```
Comment (caithomas6789) [inline report/report.py:699]: Good to remember to change the parameters for the functions.

File: report/report.py [hunk 11]
```diff
@@ -540,15 +779,20 @@ def handler(event=None, context=None):
...
     pdf_file_path = './Bandcamp-Daily-Report.pdf'
 
     convert_html_to_pdf(html_report, pdf_file_path)
-    send_email(connection, pdf_file_path)
+
+    t1_stop = perf_counter()
+    print("Elapsed time report 2 during the whole program in seconds:",
+          t1_stop-t1_start)
+
+    # send_email(connection, pdf_file_path)
```
Comment (caithomas6789) [inline report/report.py]: Is this meant to be commented out ?

File: report/static/style.css [hunk 12]
```diff
@@ -76,7 +76,7 @@ th, td {
   border-bottom: 1px solid #011022;
 }
 th{
-  background-color: #AFEEEE;
+  background-color: #62929b;
```
Comment (caithomas6789) [inline report/static/style.css:79]: I like those colour changes.

**Review by caithomas6789 — APPROVED** [review]

File: report/report.py (hunk 11 above)
Comment (IshikaM19) [reply report/report.py]: Ahhh...I had it commented out so I didn't send emails when we exceeded our limit 😂

**Pull request merged**